WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -569,6 +576,22 @@ extension OrderDetailsViewModel {
         }
     }
 
+    func startAddressValidation(shippingLabelPluginIsActive: Bool) {
+        guard let orderShippingAddress = order.shippingAddress, orderShippingAddress != Address.empty else {
+            return
+        }
+
+        let orderID = order.orderID
+
+        addressValidator.validate(address: orderShippingAddress, onlyLocally: !shippingLabelPluginIsActive) { result in
+            guard let error = result.failure else {
+                return
+            }
+
+            ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.addressValidationFailed(error: error, orderID: orderID))
```

Comment:
It is a bit confusing to see `guard` expecting error, while usually it's for filtering out failure cases.
On a first look I expect `else { ... }` to have failure handling and rest of closure - success path.

Maybe simpler `if let error { }` will be more decriptive here?

Reply:
Oh, that makes perfect sense! Thanks for this suggestion! Adjusted here: https://github.com/woocommerce/woocommerce-ios/pull/7587/commits/7e9c1eacbe3739d08ac077dd9132132675a2b83d